src/main/java/com/project/poorlex/api/service/BattleService.java

```diff
@@ -74,4 +84,85 @@ public BattleSearchResponse searchBattle(BattleFilter filter) {
 
         return BattleSearchResponse.from(battleList);
     }
+
+    // 배틀 방 상세 조회
+    public BattleDetailResponse searchBattleDetail(Long battleRoomId){
+
+        Optional<Battle> battleOptional = battleRepository.findById(battleRoomId);
+
+        List<BattleUser> battleUsers = battleUserRepository.findByBattleId(battleRoomId);
+
+        if(battleOptional.isEmpty()){
+            throw new BattleCustomException(BattleErrorCode.BATTLE_NOT_FOUND);
+        }
+
+        Battle battle = battleOptional.get();
+
+        BattleDetailResponse response = new BattleDetailResponse();
+        response.setId(battle.getId());
+        response.setName(battle.getName());
+        response.setBudget(battle.getBudget());
+        response.setTotal(battle.getTotal());
```

Comment:
optionalBattle.get()와 같은 if문으로 분기하기보단 아래처럼 나타내는게 더 직관적입니다!
Response도 생성자가 아닌 BattleDetailResponse와 같이 Builder로 컨벤션 맞춰주시는게 좋아보이네요
```suggestion
    public BattleDetailResponse searchBattleDetail(Long battleRoomId) {

        Battle battle = battleRepository.findById(battleRoomId)
            .orElseThrow(() -> new BattleCustomException(BattleErrorCode.BATTLE_NOT_FOUND));

        List<BattleUser> battleUsers = battleUserRepository.findByBattleId(battleRoomId);

        BattleDetailResponse response = BattleDetailResponse.builder();
           .id(battle.getId());
           .name(battle.getName());
           .budget(battle.getBudget());
           .total(battle.getTotal())
           .build();
```